paymentsheet/src/main/java/com/stripe/android/paymentsheet/viewmodels/PrimaryButtonUiStateMapper.kt

```diff
@@ -0,0 +1,81 @@
+package com.stripe.android.paymentsheet.viewmodels
+
+import android.content.Context
+import com.stripe.android.paymentsheet.PaymentSheet
+import com.stripe.android.paymentsheet.R
+import com.stripe.android.paymentsheet.model.PaymentSelection
+import com.stripe.android.paymentsheet.navigation.PaymentSheetScreen
+import com.stripe.android.paymentsheet.ui.PrimaryButton
+import com.stripe.android.ui.core.Amount
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.combine
+
+internal class PrimaryButtonUiStateMapper(
+    private val context: Context,
+    private val config: PaymentSheet.Configuration?,
+    private val isProcessingPayment: Boolean,
+    private val onClick: () -> Unit,
+) {
+
+    fun forCompleteFlow(
+        currentScreenFlow: Flow<PaymentSheetScreen>,
+        buttonsEnabledFlow: Flow<Boolean>,
+        amountFlow: Flow<Amount?>,
```

Comment:
I'm wondering if it would be cleaner to pass the other parameters through the constructor, since they're shared between both methods.

Reply:
Sure, I can do that!